### PR TITLE
Added Player Avatar Overlay to Lobby

### LINKED ISSUE #317: Redesign of how roles are revealed to the detective, and how a player sees their own role w/ hats on avatars
### What are you proposing?

- Players can see what their role is by their avatar having the associated hat, and when the detective gets to see a role their icon is revealed to them

### Is this ready to implement?
Yes:

#### Screenshots
Seeing your own role throughout the whole game:
Game Screen (Role Reveal)
Reveal of role to the detective:
Game Screen (Role Reveal) (1)

Figma:
https://www.figma.com/file/zoB3yUb1s6WonkGmRZcRYR/MAFIA-Assignment-2?node-id=309%3A259

#### Assets
All the different avatars as svg (use depending on how its implemented in #305, likely use avatars without face)
[Avatars without face.zip](https://github.com/Mafia-Hands/Mafia/files/6289105/Avatars.without.face.zip)
[Avatars with smiley face.zip](https://github.com/Mafia-Hands/Mafia/files/6289106/Avatars.with.smiley.face.zip)
[Avatars dead face.zip](https://github.com/Mafia-Hands/Mafia/files/6289107/Avatars.dead.face.zip)

### Redesign approval:

- [x] Redesign approved by: Daniel Etzinger

### PR DESCRIPTION
**Pull request by ODE-BOY**
The image of the players roll will be overlayed onto their position on the page.
If a detective respects a mafia the mafia image will be overlayed onto the suspected player.

# Description
This Draft PR is Fixing #317 

This PR will overlay a characters hat avatar over the player's respective avatar.
It will also overlay the mafia character over the detectives suspect if said suspect is a mafia.

# Demo

Before: [Insert Screenshot]
image

After: [Insert Screenshot]
image
image
image
image

# Type of change:
* New feature (Non-breaking change that adds functionality)

# Checklist:
* Have you merged main into your branch?
* Have you tested your changes to ensure it works as expected and does not break existing functionality?
* If applicable, please ensure sufficient tests are added that is related to the changes.
* Please ensure this PR has a `label`, is linked to an `issue` and is related to a `project`
* If new documentation is required for this change, have you created a new documentation issue that describes the documentation needed?

### CODE REVIEW

**Comment by JChamberlain28** [comment]
based on the approach taken for #305 you might want to only overlay the hats onto the existing avatars :)

**Review by JChamberlain28 — CHANGES_REQUESTED** [review]
Please see the closed PR #336 . That does not use overlaying for avatars. This impacts this draft PR. A decision must be made on if this work is to be dropped or continued due to the time constraint. Excellent work so far though.

File: frontend/src/Components/Player.jsx [hunk 1]
```diff
@@ -62,6 +67,15 @@ export default function Player({ playerName, style, childRef }) {
...
 
+    // apply player avatar hat depending on what role the player has
+    const playerHat = classNames({
+        [citizenImage]: (gameState.role === 'civilian'),
+        [detectiveImage]: (gameState.role === 'detective'),
+        [jesterImage]: (gameState.role === 'jester'),
+        [mafiaImage]: (gameState.role === 'mafia'),
+        [medicImage]: (gameState.role === 'medic'),
+    })
+
```
Comment (JChamberlain28) [inline frontend/src/Components/Player.jsx:78]: This is a really clean way of doing it. nice!

**Review by JChamberlain28 — APPROVED** [review]
Tested and it works on my pc with 1080p and normal display scaling. Might be worth someone testing on a high DPI screen e.g. a mac or high end windows laptop.

The approach taken is reasonable.
I appreciate the hard work and fast action, bearing in mind this issue was blocked for a bit. 

Good work!

**Review by AngelaLH — APPROVED** [review]
Looks awesome and matches the design :D There are some instances where it may be appropriate to have the hats showing e.g. on the end screen when the jester won they didn't have their hat, but we can make that a future issue as it wasn't stated in this one.  It's working on my 4K screen but haven't tested on a larger monitor. Thanks for your hard work!
chrome_OKpht9Ynij

**Review by AngelaLH — COMMENTED** [review]
Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)

**Comment by JChamberlain28** [comment]
> Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)

This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.

**Comment by ODE-BOY** [comment]
> 
> 
> > Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)
> 
> This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.



> 
> 
> > Just a quick thing - on the role reveal screen only the citizen has a smiley face. Was this the case before your changes? Might want to remove your changes to Medic.svg etc as they are not used in your code, otherwise, all is good :)
> 
> This applies to issue #315, the PR for that issue shows a jester with a smiley face so I think this change comes in this PR, #335.

Does this mean this is safe to merge?

**Pull request merged**